<a href="https://colab.research.google.com/github/ishita164/DataAnalytics/blob/Experiments/DA2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exp2: Analyze statistical data using Python/R programming

Objective:

Perform statistical data analysis such as: Estimators of the main statistical measures (mean, variance, standard deviation, covariance correlation, standard error), Main distributions ( Normal distribution, chi-square distribution), Hypothesis testing, pair-wise association (Pearson correlation test, t-test, ANOVA), Non-parametric test (Spearman rank0

DataSet:

Latest country level no. of covid cases in world which contain Country Name, Confirmed Cases, Recovered Cases, Deaths,Active Cases, etc

Implementation:

In [3]:
import pandas as pd
import numpy as np
data = pd.read_csv('country_wise_latest.csv')
data.info()
len(data)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 187 entries, 0 to 186
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Country/Region          187 non-null    object 
 1   Confirmed               187 non-null    int64  
 2   Deaths                  187 non-null    int64  
 3   Recovered               187 non-null    int64  
 4   Active                  187 non-null    int64  
 5   New cases               187 non-null    int64  
 6   New deaths              187 non-null    int64  
 7   New recovered           187 non-null    int64  
 8   Deaths / 100 Cases      187 non-null    float64
 9   Recovered / 100 Cases   187 non-null    float64
 10  Deaths / 100 Recovered  187 non-null    float64
 11  Confirmed last week     187 non-null    int64  
 12  1 week change           187 non-null    int64  
 13  1 week % increase       187 non-null    float64
 14  WHO Region              187 non-null    ob

187

In [4]:
data.head()

Country/Region  Confirmed  ...  1 week % increase             WHO Region
0    Afghanistan      36263  ...               2.07  Eastern Mediterranean
1        Albania       4880  ...              17.00                 Europe
2        Algeria      27973  ...              18.07                 Africa
3        Andorra        907  ...               2.60                 Europe
4         Angola        950  ...              26.84                 Africa

[5 rows x 15 columns]

In [5]:
data.shape

(187, 15)

In [6]:
data.isnull()
data.isnull().sum()

Country/Region            0
Confirmed                 0
Deaths                    0
Recovered                 0
Active                    0
New cases                 0
New deaths                0
New recovered             0
Deaths / 100 Cases        0
Recovered / 100 Cases     0
Deaths / 100 Recovered    0
Confirmed last week       0
1 week change             0
1 week % increase         0
WHO Region                0
dtype: int64

In [7]:
data.describe()

Confirmed         Deaths  ...  1 week change  1 week % increase
count  1.870000e+02     187.000000  ...     187.000000         187.000000
mean   8.813094e+04    3497.518717  ...    9448.459893          13.606203
std    3.833187e+05   14100.002482  ...   47491.127684          24.509838
min    1.000000e+01       0.000000  ...     -47.000000          -3.840000
25%    1.114000e+03      18.500000  ...      49.000000           2.775000
50%    5.059000e+03     108.000000  ...     432.000000           6.890000
75%    4.046050e+04     734.000000  ...    3172.000000          16.855000
max    4.290259e+06  148011.000000  ...  455582.000000         226.320000

[8 rows x 13 columns]

In [8]:
df = pd.DataFrame(data)
print (df.sum())

Country/Region            AfghanistanAlbaniaAlgeriaAndorraAngolaAntigua ...
Confirmed                                                          16480485
Deaths                                                               654036
Recovered                                                           9468087
Active                                                              6358362
New cases                                                            228693
New deaths                                                             5415
New recovered                                                        174623
Deaths / 100 Cases                                                   564.65
Recovered / 100 Cases                                              12121.44
Deaths / 100 Recovered                                                  inf
Confirmed last week                                                14713623
1 week change                                                       1766862
1 week % inc

In [9]:
print((df['Deaths'].sum() / df['Confirmed'].sum()) * 100)
print((df['Recovered'].sum() / df['Confirmed'].sum()) * 100)


3.968548255709708
57.450293483474546


Around 4% of people die due to covid-19 who have tested covid-19 positive and 57.5% of people recovered from covid-19 who have tested covid-19 positive. 


In [10]:
print (df.count())

Country/Region            187
Confirmed                 187
Deaths                    187
Recovered                 187
Active                    187
New cases                 187
New deaths                187
New recovered             187
Deaths / 100 Cases        187
Recovered / 100 Cases     187
Deaths / 100 Recovered    187
Confirmed last week       187
1 week change             187
1 week % increase         187
WHO Region                187
dtype: int64


In [11]:
# Calculating Percentage
df['Recovered'] = (df['Recovered'] / df['Recovered'].sum()) * 100
p=df['Recovered'].max()
print(p)

df_new = df[df['Recovered'] == p ]
print(df_new)


19.503844863275972
   Country/Region  Confirmed  ...  1 week % increase  WHO Region
23         Brazil    2442375  ...              15.28    Americas

[1 rows x 15 columns]


Maximum number fo people recovered is 19.50% which is from Brazil Country.

In [12]:
print("Mean")
print("Confirmed Cases:",data['Confirmed'].mean())
print("Deaths:",data['Deaths'].mean())
print("Recovered Cases:",data['Recovered'].mean())
print("Active Cases:",data['Active'].mean())


Mean
Confirmed Cases: 88130.935828877
Deaths: 3497.51871657754
Recovered Cases: 50631.48128342246
Active Cases: 34001.935828877


In [13]:
print("Median")
print("Confirmed Cases:",data['Confirmed'].median())
print("Deaths:",data['Deaths'].median())
print("Recovered Cases:",data['Recovered'].median())
print("Active Cases:",data['Active'].median())


Median
Confirmed Cases: 5059.0
Deaths: 108.0
Recovered Cases: 2815.0
Active Cases: 1600.0


In [14]:
from statistics import mode
print("Mode:")
print("Confirmed Cases:",data['Confirmed'].mode())
# print("Confirmed Cases:",mode(data['Confirmed']))
print("Deaths:",data['Deaths'].mode())
print("Recovered Cases:",data['Recovered'].mode())
print("Active Cases:",data['Active'].mode())


Mode:
Confirmed Cases: 0       24
1       86
2    10621
dtype: int64
Deaths: 0    0
dtype: int64
Recovered Cases: 0    0
dtype: int64
Active Cases: 0    0
dtype: int64


In [15]:
print(data['Recovered'].min())
print(data['Recovered'].max())

0
1846641


In [16]:
dif = data['Confirmed'] - data['Confirmed'].mean()

In [17]:
var =(dif**2).mean()
print("Variance: ",var)
sd = var ** (1/2)
print("Standard Deviation: ",sd)

Variance:  146147459013.93173
Standard Deviation:  382292.3737323722


In [18]:
print("Variance")
print("Confirmed Cases:",data['Confirmed'].var())
print("Deaths:",data['Deaths'].var(ddof = 0))
print("Recovered Cases:",data['Recovered'].var(ddof = 0))
print("Active Cases:",data['Active'].var(ddof = 0))


Variance
Confirmed Cases: 146933198040.88837
Deaths: 197746914.53841975
Recovered Cases: 35978116744.5491
Active Cases: 45264697655.74989


In [19]:
print("Variance")
print("Confirmed Cases:",data['Confirmed'].std())
print("Deaths:",data['Deaths'].std(ddof = 0))
print("Recovered Cases:",data['Recovered'].std(ddof = 0))
print("Active Cases:",data['Active'].std(ddof = 0))


Variance
Confirmed Cases: 383318.66383061546
Deaths: 14062.251403613142
Recovered Cases: 189678.98340235036
Active Cases: 212755.01793318507


In [20]:
data.skew(axis = 0, skipna = True)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  """Entry point for launching an IPython kernel.


Confirmed                  8.725676
Deaths                     7.464481
Recovered                  6.983644
Active                    12.182067
New cases                  7.720320
New deaths                 5.970033
New recovered              6.769567
Deaths / 100 Cases         3.352173
Recovered / 100 Cases     -0.823366
Deaths / 100 Recovered          NaN
Confirmed last week        8.865198
1 week change              7.692012
1 week % increase          6.114613
dtype: float64

In [21]:
print("Q2 quantile of arr : ", np.quantile(data['Confirmed'], .50))
print("Q1 quantile of arr : ", np.quantile(data['Confirmed'], .25))
print("Q3 quantile of arr : ", np.quantile(data['Confirmed'], .75))
print("100th quantile of arr : ", np.quantile(data['Confirmed'], .1)) 

Q2 quantile of arr :  5059.0
Q1 quantile of arr :  1114.0
Q3 quantile of arr :  40460.5
100th quantile of arr :  131.00000000000003


In [22]:
data.cov()

/usr/local/lib/python3.7/dist-packages/numpy/lib/function_base.py:2536: RuntimeWarning: invalid value encountered in subtract
  X -= avg[:, None]


Confirmed  ...  1 week % increase
Confirmed               1.469332e+11  ...      -95460.699385
Deaths                  5.051853e+09  ...      -11994.562106
Recovered               6.978949e+05  ...          -0.674355
Active                  7.580405e+10  ...      -19617.630944
New cases               1.991280e+09  ...        4309.459407
New deaths              4.010829e+07  ...          74.414353
New recovered           1.382592e+09  ...        3360.460200
Deaths / 100 Cases      8.414702e+04  ...         -11.390256
Recovered / 100 Cases  -6.531143e+05  ...        -254.020816
Deaths / 100 Recovered           NaN  ...                NaN
Confirmed last week     1.295534e+11  ...     -126415.991194
1 week change           1.737976e+10  ...       30955.291809
1 week % increase      -9.546070e+04  ...         600.732146

[13 rows x 13 columns]

Covariance indicates that there is a correalation between them except death

In [23]:
data.corr()

Confirmed    Deaths  ...  1 week change  1 week % increase
Confirmed                1.000000  0.934698  ...       0.954710          -0.010161
Deaths                   0.934698  1.000000  ...       0.855330          -0.034708
Recovered                0.906377  0.832098  ...       0.910013          -0.013697
Active                   0.927018  0.871586  ...       0.847642          -0.003752
New cases                0.909720  0.806975  ...       0.959993           0.030791
New deaths               0.871683  0.814161  ...       0.894915           0.025293
New recovered            0.859252  0.765114  ...       0.954321           0.032662
Deaths / 100 Cases       0.063550  0.251565  ...       0.015095          -0.134534
Recovered / 100 Cases   -0.064815 -0.114529  ...      -0.063013          -0.394254
Deaths / 100 Recovered   0.025175  0.169006  ...      -0.013763          -0.049083
Confirmed last week      0.999127  0.939082  ...       0.941448          -0.015247
1 week change            0.954710  0.855330  ...       1.000000           0.026594
1 week % increase       -0.010161 -0.034708  ...       0.026594           1.000000

[13 rows x 13 columns]

In [24]:
s=data['Confirmed'].std()/(len(data))**1/2
s

1024.9162134508435

In [25]:
mean = data["Confirmed"].mean()
std = data["Confirmed"].std()

In [26]:
from scipy.stats import ttest_1samp

In [27]:
tset, pval = ttest_1samp(data['Confirmed'], 1000)
print("p-values",pval)

tset, pval = ttest_1samp(data['Confirmed'], 88000)
print("p-values",pval)


p-values 0.002176705169686828
p-values 0.9962780195570828


In [28]:
print("p-values",pval)
if pval < 0.05:    # alpha value is 0.05 or 5%
   print(" we are rejecting null hypothesis")
else:
  print("we are accepting null hypothesis")

p-values 0.9962780195570828
we are accepting null hypothesis


**Conclusion**

Latest country level no. of covid cases in world which contain Country Name, Confirmed Cases, Recovered Cases, Deaths,Active Cases, etc.

*   Around 4% of people die due to covid-19 who have tested covid-19 positive.

*   57.5% of people recovered from covid-19 who have tested covid-19 positive. 

*   Maximum number fo people recovered is 19.50% which is from Brazil Country.

*   Covariance indicates that there is a correalation between them except death


The smaller the p-value, the stronger the evidence that you should reject the null hypothesis. A p-value less than 0.05 (typically ≤ 0.05) is statistically significant. It indicates strong evidence against the null hypothesis, as there is less than a 5% probability the null is correct (and the results are random).




